### ISSUE TITLE
System.OutOfMemoryException: JIT Cache exhausted

### ISSUE THREAD

**Issue by mengerpiao** [Ryujinx/Ryujinx#2170]
### What's the issue you encountered?

MHR

It will crash after 3-4 hours of play 

### Log file

02:25:19.460 |S| HLE.OsThread.5 ServiceNs CountAddOnContent: Stubbed. {aocCount: 10 ; runtimeAddOnContentInstall: 0}
02:25:19.460 |S| HLE.OsThread.5 ServiceNs ListAddOnContent: Stubbed. {bufferSize: 256 ; startIndex: 0 ; aocCount: 10}
02:25:20.065 |E| HLE.GuestThread.33 Application : Unhandled exception caught: System.OutOfMemoryException: JIT Cache exhausted.
   at ARMeilleure.Translation.Cache.JitCache.Allocate(Int32 codeSize) in C:\Users\TheSkylake\Documents\GitHub\Ryujinx\ARMeilleure\Translation\Cache\JitCache.cs:line 118
   at ARMeilleure.Translation.Cache.JitCache.Map(CompiledFunction func) in C:\Users\TheSkylake\Documents\GitHub\Ryujinx\ARMeilleure\Translation\Cache\JitCache.cs:line 51
   at ARMeilleure.Translation.Translator.Translate(IMemoryManager memory, JumpTable jumpTable, UInt64 address, ExecutionMode mode, Boolean highCq) in C:\Users\TheSkylake\Documents\GitHub\Ryujinx\ARMeilleure\Translation\Translator.cs:line 246
   at ARMeilleure.Translation.Translator.GetOrTranslate(UInt64 address, ExecutionMode mode, Boolean hintRejit) in C:\Users\TheSkylake\Documents\GitHub\Ryujinx\ARMeilleure\Translation\Translator.cs:line 180
   at ARMeilleure.Instructions.NativeInterface.GetFunctionAddressWithHint(UInt64 address, Boolean hintRejit) in C:\Users\TheSkylake\Documents\GitHub\Ryujinx\ARMeilleure\Instructions\NativeInterface.cs:line 245
   at ARMeilleure.Translation.Translator.Execute(ExecutionContext context, UInt64 address) in C:\Users\TheSkylake\Documents\GitHub\Ryujinx\ARMeilleure\Translation\Translator.cs:line 142
   at Ryujinx.HLE.HOS.Kernel.Threading.KThread.ThreadStart() in C:\Users\TheSkylake\Documents\GitHub\Ryujinx\Ryujinx.HLE\HOS\Kernel\Threading\KThread.cs:line 1031
   at System.Threading.ThreadHelper.ThreadStart_Context(Object state)

### Environment?

- Ryujinx version: 1.0.X</br>
1.0.6817
- Game version: MHR 1.0

- System Specs:
  - OS: Windows 10
  - CPU: 10700
  - GPU:  RTX 980ti
  - RAM:  16GB

**Comment by CrusadingNinja** [comment]
I believe #2168 intends to resolve this.

**Comment by mengerpiao** [comment]
> I believe #2168 intends to resolve this.

No effect, the same problem

**Comment by zeroabc** [comment]
01:06:16.557 |E| HLE.GuestThread.36 Application : Unhandled exception caught: System.OutOfMemoryException: JIT Cache exhausted.
   at ARMeilleure.Translation.Cache.JitCache.Allocate(Int32 codeSize) in C:\project\ryujinx\ARMeilleure\Translation\Cache\JitCache.cs:line 118
   at ARMeilleure.Translation.Cache.JitCache.Map(CompiledFunction func) in C:\project\ryujinx\ARMeilleure\Translation\Cache\JitCache.cs:line 51
   at ARMeilleure.Translation.Translator.Translate(IMemoryManager memory, JumpTable jumpTable, UInt64 address, ExecutionMode mode, Boolean highCq) in C:\project\ryujinx\ARMeilleure\Translation\Translator.cs:line 246
   at ARMeilleure.Translation.Translator.GetOrTranslate(UInt64 address, ExecutionMode mode, Boolean hintRejit) in C:\project\ryujinx\ARMeilleure\Translation\Translator.cs:line 180
   at ARMeilleure.Instructions.NativeInterface.GetFunctionAddressWithHint(UInt64 address, Boolean hintRejit) in C:\project\ryujinx\ARMeilleure\Instructions\NativeInterface.cs:line 245
   at ARMeilleure.Translation.Translator.Execute(ExecutionContext context, UInt64 address) in C:\project\ryujinx\ARMeilleure\Translation\Translator.cs:line 142
   at Ryujinx.HLE.HOS.Kernel.Threading.KThread.ThreadStart() in C:\project\ryujinx\Ryujinx.HLE\HOS\Kernel\Threading\KThread.cs:line 1031
   at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()
  out of memory

**Comment by trihardseven** [comment]
00:50:43.899 |W| HLE.OsThread.5 ServiceNs CountAddOnContent: RuntimeAddOnContentInstall is true. Some DLC may be missing
00:50:43.899 |S| HLE.OsThread.5 ServiceNs CountAddOnContent: Stubbed. {aocCount: 28 ; runtimeAddOnContentInstall: 1}
00:50:43.900 |S| HLE.OsThread.5 ServiceNs ListAddOnContent: Stubbed. {bufferSize: 256 ; startIndex: 0 ; aocCount: 28}
00:50:44.432 |I| Ptc Save: Saved Profiling Info (size: 393649 bytes, profiled functions: 117996).
00:50:44.892 |E| CPU.BackgroundTranslatorThread.0 Application : Unhandled exception caught: System.OutOfMemoryException: JIT Cache exhausted.
   at ARMeilleure.Translation.Cache.JitCache.Allocate(Int32 codeSize) in C:\project\ryujinx\ARMeilleure\Translation\Cache\JitCache.cs:line 118
   at ARMeilleure.Translation.Cache.JitCache.Map(CompiledFunction func) in C:\project\ryujinx\ARMeilleure\Translation\Cache\JitCache.cs:line 51
   at ARMeilleure.Translation.Translator.Translate(IMemoryManager memory, JumpTable jumpTable, UInt64 address, ExecutionMode mode, Boolean highCq) in C:\project\ryujinx\ARMeilleure\Translation\Translator.cs:line 246
   at ARMeilleure.Translation.Translator.TranslateStackedSubs() in C:\project\ryujinx\ARMeilleure\Translation\Translator.cs:line 69
   at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()
[Ryujinx_1.0.0-ldn2.2_2021-04-17_11-19-13.log](https://github.com/Ryujinx/Ryujinx/files/6329512/Ryujinx_1.0.0-ldn2.2_2021-04-17_11-19-13.log)

**Comment by manfred-exz** [comment]
Same here.
Ryujinx 1.0.6843
Win10 (ver. 21354) 
32G physical memory 
System-Hacks-Expand DRAM size to 6GB enabled

```
00:19:07.527 |S| HLE.OsThread.5 ServiceNs CountAddOnContent: Stubbed. {aocCount: 28 ; runtimeAddOnContentInstall: 1}
00:19:07.527 |S| HLE.OsThread.5 ServiceNs ListAddOnContent: Stubbed. {bufferSize: 256 ; startIndex: 0 ; aocCount: 28}
00:19:07.807 |E| CPU.BackgroundTranslatorThread.1 Application : Unhandled exception caught: System.OutOfMemoryException: JIT Cache exhausted.
   at ARMeilleure.Translation.Cache.JitCache.Allocate(Int32 codeSize) in C:\projects\ryujinx\ARMeilleure\Translation\Cache\JitCache.cs:line 118
   at ARMeilleure.Translation.Cache.JitCache.Map(CompiledFunction func) in C:\projects\ryujinx\ARMeilleure\Translation\Cache\JitCache.cs:line 51
   at ARMeilleure.Translation.Translator.Translate(IMemoryManager memory, JumpTable jumpTable, EntryTable`1 countTable, UInt64 address, ExecutionMode mode, Boolean highCq) in C:\projects\ryujinx\ARMeilleure\Translation\Translator.cs:line 273
   at ARMeilleure.Translation.Translator.TranslateStackedSubs() in C:\projects\ryujinx\ARMeilleure\Translation\Translator.cs:line 76
   at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()
```

**Comment by rtanabe999** [comment]
Right Click Menu > Cache Management > Open PPTC Directory
Delete all directory

**Comment by mengerpiao** [comment]
> Right Click Menu > Cache Management > Open PPTC Directory
> Delete all directory

If you also have this symptom, you will know that deleting PPTC is useless

**Comment by gdkchan** [comment]
I will close this since it has been largely addressed by #2286 and other JIT improvements, and we didn't receive reports of the issue in a long time on this game, so I believe no longer happens on this specific title.
